helios/pipeViewer/argos_dumper/DatabaseDump/Database_dumper.cpp

```diff
@@ -26,164 +26,146 @@ namespace pipeViewer
-        {
-            genericTransactionHandler<instruction_t>(t, &printInst);
-        }
-
-        static void printMemOp(memoryoperation_t *t, std::ostream & os)
-        {
-            os << std::setbase(10);
-            os << "*memop* " << t->transaction_ID << " @ " << t->location_ID << " start: " << t->time_Start << " end: "<<t->time_End;
-            os << std::setbase(16) << std::showbase << " vaddr: " << t->virtual_ADR;
-            os << " real_addr: " << t->real_ADR << std::endl;
-        }
-
-        virtual void foundMemRecord(memoryoperation_t*t)
-        {
-            genericTransactionHandler<memoryoperation_t>(t, printMemOp);
-        }
-
-        static void printPairOp(pair_t * p, std::ostream & os){
-            os << "*pair* ";
-            for (uint32_t i = 0; i < p->nameVector.size(); ++i) {
-                os << p->nameVector[i] << "(" << p->stringVector[i] << ") ";
             }
-            os << std::endl;
-        }
 
-        virtual void foundPairRecord(pair_t * t){
-            genericTransactionHandler<pair_t>(t, printPairOp);
-        }
+            static void printInst(instruction_t *t, std::ostream & os)
+            {
+                os << std::setbase(10);
+                os << "*instruction* " << t->transaction_ID << " @ " << t->location_ID << " start: " << t->time_Start << " end: "<<t->time_End;
+                os << " opcode: " << std::setbase(16) << std::showbase << t->operation_Code << " vaddr: " << t->virtual_ADR;
+                os << " real_addr: " << t->real_ADR << std::endl;
+            }
 
-        static void printAnnotation(annotation_t* , std::ostream & )
-        {
-        }
+            virtual void foundInstRecord(instruction_t*t)
+            {
+                genericTransactionHandler<instruction_t>(t, &printInst);
+            }
 
-        virtual void foundAnnotationRecord(annotation_t*t)
-        {
-            genericTransactionHandler<annotation_t>(t, printAnnotation);
-        }
+            static void printMemOp(memoryoperation_t *t, std::ostream & os)
+            {
+                os << std::setbase(10);
+                os << "*memop* " << t->transaction_ID << " @ " << t->location_ID << " start: " << t->time_Start << " end: "<<t->time_End;
+                os << std::setbase(16) << std::showbase << " vaddr: " << t->virtual_ADR;
+                os << " real_addr: " << t->real_ADR << std::endl;
+            }
 
-        public:
-            // Sets merge/sort flags
-            void setFlags(bool merge, bool sort)
+            virtual void foundMemRecord(memoryoperation_t*t)
             {
-                merge_transactions = merge;
-                sort_by_end_time = sort;
+                genericTransactionHandler<memoryoperation_t>(t, printMemOp);
             }
 
+            static void printPairOp(pair_t * p, std::ostream & os){
+                os << "*pair* ";
+                for (uint32_t i = 0; i < p->nameVector.size(); ++i) {
```

Comment:
Can use a range based for loop

Reply:
Sadly, no. We're iterating over 2 vectors in that loop.